modules/core/src/norm.cpp

```diff
@@ -1171,7 +1171,7 @@ double norm( InputArray _src1, InputArray _src2, int normType, InputArray _mask
         // special case to handle "integer" overflow in accumulator
         const size_t esz = src1.elemSize();
         const int total = (int)it.size;
-        const int intSumBlockSize = normType == NORM_L1 && depth <= CV_8S ? (1 << 23) : (1 << 15);
+        const int intSumBlockSize = (normType == NORM_L1 && depth <= CV_8S ? (1 << 23) : (1 << 15))/cn;
         const int blockSize = std::min(total, intSumBlockSize);
         int isum = 0;
```

Comment:
> (1 << 15)

Parent of the mentioned commit contains the similar code.

The change which introduces regression is:

```.diff
-unsigned isum = 0;
+int isum = 0
```

`int` type is used here to follow the parameter type of used processing functions (see `NormDiffFunc`)

---

The proposed change is correct (also fixes [3.4.9](https://github.com/opencv/opencv/blame/3.4.9/modules/core/src/norm.cpp#L1182) on test with `kSize = 200`).

---

Regression is introduced here by adding of block processing: https://github.com/opencv/opencv/blame/ab0f0f26a1d82e71d756a8b08f061a69214a0f10/modules/core/src/stat.cpp#L1214